doc/src/pair_uf3.rst

```diff
@@ -0,0 +1,184 @@
+
+As an example, if a LAMMPS simulation contains 2 atom types (elements 'A' and 'B'), the pair_coeff command will be-
+
+.. code-block:: LAMMPS
+
+   pair_style uf3 3 2
+   pair_coeff 1 1 A_A
+   pair_coeff 1 2 A_B
+   pair_coeff 2 2 B_B
+   pair_coeff 3b 1 1 1 A_A_A
+   pair_coeff 3b 1 1 2 A_A_B
+   pair_coeff 3b 1 2 2 A_B_B
+   pair_coeff 3b 2 1 1 B_A_A
+   pair_coeff 3b 2 1 2 B_A_B
+   pair_coeff 3b 2 2 2 B_B_B
+
+If a value of "2" is specified in the :code:`pair_style uf3` command, only the two-body potential files are needed. For 3-body interaction the first atom type is the central atom. We recommend using the :code:`generate_uf3_lammps_pots.py` script (found `here <https://github.com/uf3/uf3/tree/master/lammps_plugin/scripts>`_) for generating the UF3 LAMMPS potential files from the UF3 JSON potentials.
+
+LAMMPS wild-card character "*" can also be used to specify a single UF3 LAMMPS potential file for multiple interaction. For example-
+
+.. code-block:: LAMMPS
+
+   pair_style uf3 3 2
+   pair_coeff * * A_A
+   pair_coeff 3b 1 * * A_A_A
+   pair_coeff 3b 2 * * B_B_B
+
+The file A_A will be used for 2-body interaction between atom types 1-1, 1-2 and 2-2; file A_A_A will be used 3-body interaction for atom types 1-1-1, 1-1-2, 1-2-2; and so on. Note, using a single interaction file for all types of interactions is **not** the recommended way of using :code:`pair_style uf3` and will often lead to **incorrect results**.
+
+
+UF3 LAMMPS potential files in the *potentials* directory of the LAMMPS distribution have a ".uf3" suffix. All UF3 LAMMPS potential files should start with :code:`#UF3 POT` and end with :code:`#` characters. Following shows the format of a generic 2-body UF3 LAMMPS potential file-
+
+.. code-block:: LAMMPS
+
+   #UF3 POT
+   2B LEADING_TRIM TRAILING_TRIM
+   Rij_CUTOFF NUM_OF_KNOTS
+   BSPLINE_KNOTS
+   NUM_OF_COEFF
+   COEFF
+   #
+
+The second line indicates whether the potential file contains data for 2-body (:code:`2B`) or 3-body (:code:`3B`) interaction. This is followed by :code:`LEADING_TRIM` and :code:`TRAILING_TRIM` number on the same line. The current implementation is only tested for :code:`LEADING_TRIM=0` and :code:`TRAILING_TRIM=3`. If other values are used LAMMPS is terminated after issuing an error message. The :code:`Rij_CUTOFF` sets the 2-body cutoff for the interaction described by the potential file. :code:`NUM_OF_KNOTS` is the number of knots (or the length of the knot vector) present on the very next line. The :code:`BSPLINE_KNOTS` line should contain all the knots in ascending order. :code:`NUM_OF_COEFF` is the number of coefficients in the :code:`COEFF` line. All the numbers in the BSPLINE_KNOTS and COEFF line should be space-separated.
+
+The format of a generic 3-body UF3 LAMMPS potential file is as follow-
+
+.. code-block:: LAMMPS
+
+   #UF3 POT
```

Comment:
The potential files should have metadata tags that indicate the source, a reference, the data, and in particular can indicate the units, so that LAMMPS will refused to read potential files created for incompatible units.
You are already using `utils::open_potential()` so this check will be automatic, provided the metadata tags are present.